src/components/auction/OrderPlacement/index.tsx

```diff
@@ -221,6 +224,44 @@ const OrderPlacement: React.FC<OrderPlacementProps> = (props) => {
     }
   }, [onUserPriceInput, price, derivedAuctionInfo, showPriceInverted])
 
+  useEffect(() => {
+    const coversClearingPrice = (price: string | undefined, lessThan: boolean): boolean => {
```

Comment:
You are introducing quite some logic.

Please take a look at: `useGetOrderPlacementError`

You can reuse here:
```
const price = showPricesInverted
    ? getInverse(Number(priceFromState), NUMBER_OF_DIGITS_FOR_INVERSION).toString()
    : priceFromState

const { buyAmountScaled, sellAmountScaled } = convertPriceIntoBuyAndSellAmount(
    derivedAuctionInfo?.auctioningToken,
    derivedAuctionInfo?.biddingToken,
    price == '-' ? '1' : price,
    sellAmount,
  )
  ```
  ```
  if (
    derivedAuctionInfo?.clearingPriceSellOrder != null &&
    derivedAuctionInfo?.clearingPrice != null &&
    derivedAuctionInfo?.auctioningToken != undefined &&
    derivedAuctionInfo?.biddingToken != undefined &&
    auctionState == AuctionState.ORDER_PLACING &&
    buyAmountScaled &&
    sellAmountScaled
      ?.mul(derivedAuctionInfo?.clearingPriceSellOrder?.buyAmount.raw.toString())
      .lte(
        buyAmountScaled.mul(derivedAuctionInfo?.clearingPriceSellOrder?.sellAmount.raw.toString()),
      )
  ) {
    error =
      error ?? showPricesInverted
        ? 'Price must be lower than ' + derivedAuctionInfo?.clearingPrice?.invert().toSignificant(5)
        : 'Price must be higher than ' + derivedAuctionInfo?.clearingPrice?.toSignificant(5)
  }
  ```

Reply:
I find the lessThan hard to read.

I prefer standardizing everything by inverting the price to a standardizedPrice variable and then always doing the same check

Reply:
I have not yet identified how I can reuse `useGetOrderPlacementError`. (I will check it) 

Ok, you mean name the variable with the same name so I can keep track of it. For example [`invertedPrice`](https://github.com/gnosis/ido-ux/pull/545/files#diff-5f88dd7ddd21194834d24abc9a7dcb126132d27605fec3cf6695c8228f9a7b07R267) as `showPricesInverted`